source/agora/common/Transaction.d

```diff
@@ -260,6 +260,34 @@ public bool verifyData (Transaction tx, Output* delegate (Hash hash, size_t inde
     }
 }
 
+/*******************************************************************************
+
+    Get result of signature verification
+
+    Params:
+        tx = `Transaction`
+        findOutput = delegate for finding `Output`
+
+    Return:
+        Return true if this signature is verified.
+
+*******************************************************************************/
+
+public bool verifySignature (Transaction tx, Output* delegate (Hash hash, size_t index) findOutput)
+{
+    foreach (input; tx.inputs)
+    {
+       if (auto output = findOutput(input.previous,input.index))
+        {
+            if (!output.address.verify(input.signature, hashFull(input)[0..$]))
```

Comment:
That is not very secure :)

Reply:
Which part?

Reply:
Modified to sign the entire transaction object.